diff --git a/src/connected_components.h b/src/connected_components.h
index aeb1278..5aa9142 100644
--- a/src/connected_components.h
+++ b/src/connected_components.h
@@ -230,0 +231 @@ ArrayIDIDFunc compute_biconnected_components(
+		min_succ_depth[x] = std::numeric_limits<int>::max();
diff --git a/src/tiny_id_func.h b/src/tiny_id_func.h
index 7bdc9f7..5a47b50 100644
--- a/src/tiny_id_func.h
+++ b/src/tiny_id_func.h
@@ -21 +21,3 @@ public:
-		:preimage_(preimage), data_((preimage + entry_count_per_uint64 - 1) / entry_count_per_uint64){}
+		:preimage_(preimage), data_((preimage + entry_count_per_uint64 - 1) / entry_count_per_uint64){
+		data_.fill(0);
+	}
@@ -30,0 +33 @@ public:
+		data_.fill(0);
